packages/nextjs/src/index.server.ts

```diff
@@ -23,12 +23,20 @@ const domain = domainModule as typeof domainModule & { active: (domainModule.Dom
 // During build, the main process is invoked by
 //   `node next build`
 // and child processes are invoked as
-//   `node <path>/node_modules/jest-worker/build/workers/processChild.js`,
-// whereas at runtime the process is invoked as
+//   `node <path>/node_modules/.../jest-worker/processChild.js`.
+// The former is (obviously) easy to recognize, but the latter could happen at runtime as well. Fortunately, the main
+// process hits this file before any of the child processes do, so we're able to set an env variable which the child
+// processes can then check. During runtime, the main process is invoked as
 //   `node next start`
 // or
-//   `node /var/runtime/index.js`.
-const isBuild = new RegExp('build').test(process.argv.toString());
+//   `node /var/runtime/index.js`,
+// so we never drop into the `if` in the first place.
+let isBuild;
+if (process.argv.includes('build') || process.env.BUILD_PHASE) {
+  process.env.BUILD_PHASE = 'true';
```

Comment:
Could this conflict with user set environmental variables? Perhaps we append Sentry onto it?

```suggestion
  process.env.SENTRY_NEXTJS_BUILD_PHASE = 'true';
```

Reply:
Fair point, and namespacing it was actually my first instinct as well. Just took it out because it seemed a little weird given that it's not the _Sentry_ build phase, it's just the build phase, for everyone, but happy to put it back in. (I'm just going for `SENTRY` rather than `SENTRY_NEXTJS` because here, by definition, everything is nextjs.)

```suggestion
  process.env.SENTRY_BUILD_PHASE = 'true';
```